File: 1-two-sum/two-sum.py
import collections
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mydict = collections.defaultdict(int)
        for i in range(len(nums)):
            diff = target - nums[i]
            if(diff in mydict):
                return [i,mydict[diff]]
            else:
                mydict[nums[i]] = i
            
        

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hm = {}

    for(let i=0; i<nums.length; i++){
        let rem = target - nums[i]

        if(hm[rem] != undefined){
            return [i, hm[rem]]
        }

        hm[nums[i]] = i
    }

    return [-1,-1]
};